lib/core/macros.nim

```diff
@@ -509,7 +509,13 @@ proc parseStmt*(s: string): NimNode {.noSideEffect, compileTime.} =
   ## Expects one or more statements. Raises ``ValueError`` for parsing errors.
   result = internalParseStmt(s)
   let x = internalErrorFlag()
-  if x.len > 0: raise newException(ValueError, x)
+  if x.len > 0:
+    var msg = "parse error: "
+    msg.addQuoted x
+    msg.add "\ns:\n------\n"
```

Comment:
to avoid back and forth, what formatting do you suggest?
my main concern is ease of distinguishing beginning/end of that block of code from other msgs printed out in terminal

Reply:
ping @Araq 
how about following: add indentation to `s`, as follows:
```
Error: unhandled exception: internalError: "/Users/timothee/git_clone/nim/Nim/lib/core/macros.nim(515, 12) Error: invalid indentation"; parseStmt failed with: 
  import foo
  bar())
```